### PR TITLE
Fix #2262: use typed responses for REST/Async endpoints where possible

### LINKED ISSUE #2262: Improve SGv2/REST-API async endpoint signatures to be typed
(from CR of #2213)

There are benefits from returning strongly typed `Uni` responses so let's try to indicate actual response POJO type where possible.

In cases where type cannot be included, DTOs should be annotated with `@RegisterForReflection` for Graal Native image purposes.

**EDIT**: Looks like many GET methods have "wrapped or unwrapped" handling (with `raw` query parameter) which prevents full typing. But I can work on some improvements still, and add above-mentioned annotation for other types.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Improve type-safety of REST API endpoints to help with Quarkus native bindings (and documentation, general cleanliness).

NOTE: none of GET endpoints can use common base class due to odd wrapper-or-unwrapped response type (controlled by "raw" query param). Changed CREATE/UPDATE/DELETE endpoints, however, and added annotations for other response types to help with native bindings.

**Which issue(s) this PR fixes**:
Fixes #2262

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/models/Sgv2ColumnDefinition.java [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import io.quarkus.runtime.annotations.RegisterForReflection;
 import java.util.Objects;
 import org.eclipse.microprofile.openapi.annotations.media.Schema;
 
 // Copied from SGv1 ColumnDefinition
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @Schema(name = "ColumnDefinition")
+@RegisterForReflection
```
Comment (ivansenic) [inline apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/models/Sgv2ColumnDefinition.java:13]: Did you consider moving DTOs to records and getting rid of all that boilerplate code?

**Review by ivansenic — COMMENTED** [review]
Wait I am confused, all of the changes are from `RestResponse<Object>` to `RestResponse<Map>`? so no concrete types at all are returned?

File: apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/models/Sgv2ColumnDefinition.java (hunk 1 above)
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/models/Sgv2ColumnDefinition.java:13]: Yes, there is a separate issue (#2264) for that. Just need to figure out its priorization.

**Comment by tatu-at-datastax** [comment]
> Wait I am confused, all of the changes are from `RestResponse<Object>` to `RestResponse<Map>`? so no concrete types at all are returned?

Correct: as per my other note the big problem is the "raw" handling: everything is either

     Sgv2RestResponse<ReturnValueType>

OR

    ReturnValueType

because caller can specify `raw` as true (no wrapper) or `false` (use wrapper).

So the only things that are more concrete are simple "name" : "<name>" pairs.
However... now that I wrote that I realized I can make that response a Record. So will go and do that instead. Thanks!

**Comment by ivansenic** [comment]
> > Wait I am confused, all of the changes are from `RestResponse<Object>` to `RestResponse<Map>`? so no concrete types at all are returned?
> 
> Correct: as per my other note the big problem is the "raw" handling: everything is either
> 
> ```
>  Sgv2RestResponse<ReturnValueType>
> ```
> 
> OR
> 
> ```
> ReturnValueType
> ```
> 
> because caller can specify `raw` as true (no wrapper) or `false` (use wrapper).
> 
> So the only things that are more concrete are simple "name" : "" pairs. However... now that I wrote that I realized I can make that response a Record. So will go and do that instead. Thanks!

But then we did not get much from this I guess.. Map is still too unspecific imo..

**Review by ivansenic — APPROVED** [review]
Make sure tests are green..

**Comment by tatu-at-datastax** [comment]
> > > Wait I am confused, all of the changes are from `RestResponse<Object>` to `RestResponse<Map>`? so no concrete types at all are returned?
> > 
> > 
> > Correct: as per my other note the big problem is the "raw" handling: everything is either
> > ```
> >  Sgv2RestResponse<ReturnValueType>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > OR
> > ```
> > ReturnValueType
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > because caller can specify `raw` as true (no wrapper) or `false` (use wrapper).
> > So the only things that are more concrete are simple "name" : "" pairs. However... now that I wrote that I realized I can make that response a Record. So will go and do that instead. Thanks!
> 
> But then we did not get much from this I guess.. Map is still too unspecific imo..

Agreed; I had forgotten about this weird wrapping part wrt typing.

I went ahead and added "name response" type (record) to add a small amount of typing, seems like best we can do for now.
And at least there are now annotations so that type discovery should work for native images.

**Comment by tatu-at-datastax** [comment]
All tests green, will merge.

**Pull request merged**